### PR TITLE
ref: re-enable codecov with `continue-on-error` and frozen

### PR DESCRIPTION
**Pull request by asottile-sentry**

### CODE REVIEW

File: tools/pin_github_action.py [hunk 1]
```diff
@@ -1,45 +1,30 @@
...
-        )
-    except HTTPError as e:
-        print(f"Status {e.code} while resolving {ref} for {repo}.")
-        if e.code == 403:
-            print("You most likely didn't authorize your token for SAML to the getsentry org.")
-        return ref
-
-    data: dict[str, str] = json.load(resp)
-    return data["sha"]
+    cmd = ("git", "ls-remote", "--exit-code", f"https://github.com/{repo}", ref)
```
Comment (armenzg) [inline tools/pin_github_action.py:20]: Commands' help: `git-ls-remote - List references in a remote repository`

```
 --exit-code
           Exit with status "2" when no matching refs are found in the remote repository. Usually the command exits with status "0" to indicate it successfully talked with the remote
           repository, whether it found any matching refs.
```

Sample output (3 lines):
```text
4ee1af708e1e130e29bc46388c0a9192dcc02443	refs/reviewable/pr3836/r1
8850500135277823d37fbadabb04b202d738db74	refs/tags/v1.8.8
91c61b2c5cc09747a95fd1505d4df4f1f4fcf053	refs/tags/v1.8.8^{}
```

File: tools/pin_github_action.py [hunk 2]
```diff
@@ -1,45 +1,30 @@
...
+    out = subprocess.check_output(cmd)
+    for line in out.decode().splitlines():
+        sha, refname = line.split()
+        if refname in (f"refs/tags/{ref}", f"refs/heads/{ref}"):
```
Comment (armenzg) [inline tools/pin_github_action.py:24]: Does this mean that we can use a tag or a branch name?

File: tools/pin_github_action.py [hunk 3]
```diff
@@ -1,45 +1,30 @@
...
-import urllib.request
+import subprocess
 from functools import lru_cache
 from typing import Sequence
-from urllib.error import HTTPError
 
 
 @lru_cache(maxsize=None)
-def get_sha(repo: str, ref: str, github_token: str) -> str:
+def get_sha(repo: str, ref: str) -> str:
```
Comment (armenzg) [inline tools/pin_github_action.py:11]: I see you're removing the usage of `github_token`; is it because `ls-remote` solves the problem?

**Review by armenzg — APPROVED** [review]

File: tools/pin_github_action.py (hunk 3 above)
Comment (asottile-sentry) [reply tools/pin_github_action.py:11]: yeah, github actions can only be public repos -- so this is simpler and requires fewer creds

File: tools/pin_github_action.py (hunk 2 above)
Comment (asottile-sentry) [reply tools/pin_github_action.py:24]: this is mostly to filter out ls-remote's lazy matching (which includes stuff like `refs/heads/foo/v2`)

**Pull request merged**